classes/class-wc-connect-nux.php

```diff
@@ -177,15 +200,23 @@ public static function get_banner_type_to_display( $status = array() ) {
 				case self::JETPACK_ACTIVATED_NOT_CONNECTED:
 					return 'before_jetpack_connection';
 				case self::JETPACK_CONNECTED:
-					// Has the user just gone through our NUX connection flow?
-					if ( isset( $status['should_display_after_cxn_banner'] ) && $status['should_display_after_cxn_banner'] ) {
-						return 'after_jetpack_connection';
-					}
+				case self::JETPACK_DEV:
 					// Has the user already accepted our TOS? Then do nothing.
 					// Note: TOS is accepted during the after_connection banner
-					if ( isset( $status['tos_accepted'] ) && ! $status['tos_accepted'] ) {
+					if (
+						isset( $status['tos_accepted'] )
+						&& ! $status['tos_accepted']
+						&& isset( $status['can_accept_tos'] )
+						&& $status['can_accept_tos']
+					) {
 						return 'tos_only_banner';
 					}
+
+					// Has the user just gone through our NUX connection flow?
+					if ( isset( $status['should_display_after_cxn_banner'] ) && $status['should_display_after_cxn_banner'] ) {
```

Comment:
I realize you probably swapped these two if-statements because of the issue I reported yesterday. Unfortunately though this won't work. The problem is that although connecting Jetpack is considered an acceptance of the TOS, we don't actually set the `tos_accepted` option when that happens.

So, if jetpack isn't already connected, you'll see the "accept tos" screen instead of the final welcome screen.

Ideally I think we should set the `tos_accepted` option when the user installs/activates/connects Jetpack, but since we're not doing that your best bet is to swap these back to the order in which they were.